### PR TITLE
Convert existing widgets to widgetbase

### LINKED ISSUE #95: Convert existing widgets and mixins to widgetBase and `d`

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [x] Unit or Functional tests are included in the PR

**Description:**

Convert existing widgets to the new `createWidgetBase` base class.

Resolves #95

### CODE REVIEW

**Comment by agubler** [comment]
This should allow it so that the new version can just be dropped in to existing consumers

**Comment by agubler** [comment]
@kitsonk can I get a review of this at some point - this should move all existing widgets to use `createWidgetBase` and other `dojo-interfaces` API.

I imagine a lot of these will require re-working when we have a container solution in place, so didn't focus on making them "better" but just to get parity for the existing widgets.

File: src/bases/createWidgetBase.ts [hunk 1]
```diff
@@ -96,6 +96,13 @@ function manageDetachedChildren(instance: Widget<WidgetState>): void {
 	internalState.currentChildrenMap.clear();
 }
 
+function formatTagNameAndClasses(tagName: string, classes: string[]) {
```
Comment (agubler) [inline src/bases/createWidgetBase.ts]: Need to add unit tests for this.

File: src/mixins/createRenderMixin.ts [hunk 2]
```diff
@@ -1,254 +1,9 @@
...
-
-			instance.own(instance.on('state:changed', () => instance.invalidate()));
-
-			widgetClassesMap.set(instance, []);
-		}
-	})
-	.static({
-		idBase: 'widget'
-	}) as RenderMixinFactory;
+export {
```
Comment (kitsonk) [inline src/mixins/createRenderMixin.ts]: I assume the main reason we are doing this is to minimise the amount of refactoring?

File: src/mixins/createTabbedMixin.ts [hunk 3]
```diff
@@ -248,18 +245,26 @@ const createTabbedMixin: TabbedMixinFactory = createRenderMixin
 					if (tabVNode.properties) {
 						(tabVNode as any).properties['data-visible'] = String(isActiveTab);
 					}
-					childrenNodes[idx] = tabVNode;
+					const childNode = {
```
Comment (kitsonk) [inline src/mixins/createTabbedMixin.ts]: Ideally we would be explicit what the object literal type is.  It should be `DNode`, right?

Comment (agubler) [inline src/mixins/createTabbedMixin.ts]: Yeah `DNode` or `HNode` I believe

**Pull request merged**